src/setup/config.rs

```diff
@@ -18,6 +23,56 @@ pub struct NodeConfig {
     pub log_to_stdout: bool,
     /// The path of the cache directory of the node.
     pub path: PathBuf,
+    /// The network socket address of the node.
+    pub local_addr: Option<SocketAddr>,
+    /// The REST API socket address of the node.
+    pub rest_api_addr: Option<SocketAddr>,
+}
+
+impl NodeConfig {
+    /// Continuously try to read a string from a file.
+    async fn try_read_to_string(file_path: &Path) -> String {
+        loop {
+            match fs::read_to_string(&file_path) {
+                Ok(content) => return content,
+                Err(_) => {
+                    sleep(Duration::from_millis(100)).await;
+                    continue;
+                }
+            };
+        }
+    }
```

Comment:
I wanted to use async closure here, but unfortunately, those are still unstable.